membership-attribute-service/app/services/PaymentFailureAlerter.scala

```diff
@@ -125,4 +127,28 @@ object PaymentFailureAlerter extends LoggingWithLogstashFields {
 
     alertAvailable.getOrElse(Future.successful(false))
   }
+
+  def mostRecentPayableInvoicesOlderThanOneMonth(recentInvoices: List[Invoice]): List[Invoice] = {
+    implicit def localDateOrdering: Ordering[DateTime] = Ordering.fromLessThan(_ isAfter _)
+    val nonZeroInvoicesOlderThanOneMonth = recentInvoices.filter(invoice => invoice.invoiceDate.isBefore(DateTime.now().minusMonths(1)) && invoice.amount > 0)
+    nonZeroInvoicesOlderThanOneMonth.sortBy(_.invoiceDate).take(2)
+  }
+
+  def accountHasMissedRecentPayments(accountId: AccountId, isPaidSub: Boolean, recentInvoices: List[Invoice], recentPayments: List[Payment]): Boolean = {
```

Comment:
This logic is all a bit fiddly (see unit tests). The logic would be simpler if we knew whether the account had autopay enabled or not, but I can't get that info without making an extra API call and this endpoint is already making too many requests (IMO).

Reply:
are the recentInvoices all from the same subscription? or does isPaidSub mean that there is at least one paid sub in the account ?

Reply:
I find the name of this function slightly confusing because it never really checks that the payments or invoices are recent, that seems to be just a prerequisite to the input params. But it does check the opposite, that the invoices should be older than one month.

Reply:
> are the recentInvoices all from the same subscription

This should be passed the invoices and payments sections of an account summary. I didn't want to pass a big object around unnecessarily, but I can amend this if you think it'd be clearer?

Reply:
> I find the name of this function slightly confusing because it never really checks that the payments or invoices are recent, that seems to be just a prerequisite to the input params

Good point, I will drop 'recent' from the name 👍

Reply:
> > are the recentInvoices all from the same subscription
> 
> This should be passed the invoices and payments sections of an account summary. I didn't want to pass a big object around unnecessarily, but I can amend this if you think it'd be clearer?
 I was asking that because isPaidSub refers to a subscription but the invoices could potentially belong to multiple subs in the same account right ? So when isPaidSub = false does it mean all the subs in this account are not paid subs ?

Reply:
> I was asking that because isPaidSub refers to a subscription but the invoices could potentially belong to multiple subs in the same account right?

Correct.

> So when isPaidSub = false does it mean all the subs in this account are not paid subs ?

Not necessarily. Now that we support multiple subs in the same billing account I guess we should be open to the possibility that some will be free but others will be paid. The current implementation should omit any payment method details if clients ask about a free sub (based on https://github.com/guardian/membership-common/blob/aa309b43c45a3d6436eba1eb6c782f14512b04f0/src/main/scala/com/gu/memsub/services/PaymentService.scala#L27-L28). Similarly, using this boolean here allows us to omit details about a payment method reset in cases where clients are specifically asking about a free sub.

Reply:
Improved this slightly with https://github.com/guardian/members-data-api/pull/370/commits/c9add6659d515936823a357377ca8cd1a67cc8db.